File: react-shop/src/services/BrandService.js
import axios from "../api/axiosConfig";

const BrandService = {
  getAll: async () => {
    const res = await axios.get("/brands");
    return res.data;
  },
  getById: async (id) => {
    const res = await axios.get(`/brands/${id}`);
    return res.data;
  },
  create: async (data) => {
    const res = await axios.post("/brands", data);
    return res.data;
  },
  update: async (id, data) => {
    const res = await axios.put(`/brands/${id}`, data);
    return res.data;
  },
  delete: async (id) => {
    await axios.delete(`/brands/${id}`);
  },
};

export default BrandService;

File: kimanh/src/main/java/com/example/kimanh/service/BrandService.java
package com.example.kimanh.service;

import java.util.List;

import com.example.kimanh.dto.BrandDto;

public interface BrandService {
    List<BrandDto> getAll();
    BrandDto getById(Integer id);
    BrandDto create(BrandDto dto);
    BrandDto update(Integer id, BrandDto dto);
    void delete(Integer id);
}
